examples/cryptocurrency-advanced/backend/src/transactions.rs

```diff
@@ -95,27 +94,46 @@ transactions! {
 
         /// Create wallet with the given `name`.
         struct CreateWallet {
-            /// `PublicKey` of the new wallet.
-            pub_key: &PublicKey,
             /// Name of the new wallet.
             name:    &str,
         }
     }
 }
 
-impl Transaction for Transfer {
-    fn verify(&self) -> bool {
-        (self.from() != self.to()) && self.verify_signature(self.from())
+impl CreateWallet {
+    #[doc(hidden)]
+    pub fn sign(name: &str, pk: &PublicKey, sk: &SecretKey) -> Message<RawTransaction> {
+        Protocol::sign_transaction(CreateWallet::new(name), CRYPTOCURRENCY_SERVICE_ID, *pk, sk)
     }
+}
 
-    fn execute(&self, fork: &mut Fork) -> ExecutionResult {
-        let mut schema = Schema::new(fork);
+impl Transfer {
+    #[doc(hidden)]
+    pub fn sign(
+        pk: &PublicKey,
+        to: &PublicKey,
+        amount: u64,
+        seed: u64,
+        sk: &SecretKey,
+    ) -> Message<RawTransaction> {
+        Protocol::sign_transaction(Transfer::new(to, amount, seed), CRYPTOCURRENCY_SERVICE_ID, *pk, sk)
+    }
+}
+
+impl Transaction for Transfer {
+    fn execute(&self, mut context: TransactionContext) -> ExecutionResult {
+        let ref from = context.author();
+        let hash = context.tx_hash();
+
+        let mut schema = Schema::new(context.fork());
 
-        let from = self.from();
         let to = self.to();
-        let hash = self.hash();
         let amount = self.amount();
 
+        if from == to {
+            return Err(ExecutionError::new(ERROR_SENDER_SAME_AS_RECEIVER))
+        }
```

Comment:
Please remind me, are we going to retire the `verify()` method on transactions completely? So there will be no preliminary verification of transactions before they are executed?